### ISSUE TITLE
amazon-ebs builder does not accept "/" in instanceProfileName

### ISSUE THREAD

**Issue by yogeek** [hashicorp/packer#10024]
#### Overview of the Issue

When trying to create an AWS AMI with an instance profile ARN with a path (containing "/" characters) like `arn:aws-cn:iam::$ACCOUNT_ID:instance-profile/foo/bar/test-profile`, the following error occurs : 

```
Couldn't find specified instance profile: ValidationError: 1 validation error detected: Value '/foo/bar/test-profile' at 'instanceProfileName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]+
```

#### Reproduction Steps

Try to build an AWS AMI with the code provided in the **Simplified Packer Buildfile** section where `iam_instance_profile` value contains "/" characters : 
`"iam_instance_profile": "/foo/bar/test-profile",`

### Packer version

`Packer v1.6.4`

### Simplified Packer Buildfile

```
{
  "min_packer_version": "1.3.0",
  "variables": {
    "aws_region": "{{ env `AWS_DEFAULT_REGION` }}",
    "aws_profile": "{{ env `AWS_PROFILE` }}",
    [...]
  },
  "builders": [
    {
      "type": "amazon-ebs",
      "region": "{{user `aws_region`}}",
      "instance_type": "t3.micro",
      "associate_public_ip_address": true,
      "ssh_interface" : "public_ip",
      "iam_instance_profile": "/foo/bar/test-profile",
	  [...]
```

### Log Fragments and crash.log files

```
2020/10/01 16:30:13 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/10/01 16:30:13 machine readable: amazon-ebs,error []string{"Couldn't find specified instance profile: ValidationError: 1 validation error detected: Value '/foo/bar/test-profile' at 'instanceProfileName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\\w+=,.@-]+\n\tstatus code: 400, request id: bbc5a9f9-2be5-4ad1-80be-e319b953f7a5"}
	status code: 400, request id: bbc5a9f9-2be5-4ad1-80be-e319b953f7a5
==> Builds finished but no artifacts were created.
2020/10/01 16:30:13 [INFO] (telemetry) Finalizing.
Build 'amazon-ebs' errored after 20 seconds 62 milliseconds: Couldn't find specified instance profile: ValidationError: 1 validation error detected: Value '/foo/bar/test-profile' at 'instanceProfileName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]+
	status code: 400, request id: bbc5a9f9-2be5-4ad1-80be-e319b953f7a5

==> Wait completed after 20 seconds 62 milliseconds

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Couldn't find specified instance profile: ValidationError: 1 validation error detected: Value '/foo/bar/test-profile' at 'instanceProfileName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]+
	status code: 400, request id: bbc5a9f9-2be5-4ad1-80be-e319b953f7a5
```

**Comment by nywilken** [comment]
Hi there, thanks for reaching out. This error actually coming from AWS and not Packer.

With that said, are you specifying the full ARN for the instance profile name? 

If so, then you just need to specify the name of the profile, which is the last part of the ARN. 

For example given the following ARN: `arn:aws:iam::123456789:role/test-role`  the value in Packer would be just `"test-role"`; more specifically `"iam_instance_profile": "test-role"`. 

Please give this a try and let us know if you run into any further problems.

**Comment by yogeek** [comment]
Hi @nywilken and thank you for your help,

As I mentionned in my first message, I specified the last part of the ARN :

for ARN = `arn:aws-cn:iam::$ACCOUNT_ID:instance-profile/foo/bar/test-profile`
I set `"iam_instance_profile": "/foo/bar/test-profile"`

I tried by duplicating the instance profile without any path like this :

ARN = `arn:aws-cn:iam::$ACCOUNT_ID:instance-profile/test-profile`
and `"iam_instance_profile": "/foo/bar/test-profile"` is working ok

So it really seems that it does not accept an instance profile name with a path

**Comment by nywilken** [comment]
Hi @yogeek thanks for the quick response. To elaborate on my comment a bit. It is the AWS API that is erroring, which to your point instance profiles with slashes in the name don't appear to be  allowed. 

Is this a newly created instance profile?

I know that IAM profiles do have a Path field which does support values with slashes but that is different from the InstanceProfileName that Packer is expecting. 

I see from #10026 that you are working in the China region and was curious to see if validation rules for IAM profile names were different and found https://docs.amazonaws.cn/en_us/IAM/latest/APIReference/API_CreateInstanceProfile.html.  

If the instance Profile name you are providing is not in fact the profile Path and you are still able to create new instance profiles with slashes in their name we might be running into an upstream-bug or need to change the API endpoint we are hitting when querying for instance profiles in the China region.

**Comment by yogeek** [comment]
Hi @nywilken ,

Sorry, I made a typo in my previous message so I edited it. (in the second test where I duplicate the instance profile without a path, it is `"iam_instance_profile": "test-profile"`  that is working ok obviously.

To answer your questions, my instance profile has been created by terraform recently, and I can confirm you that creating an `aws_iam_instance_profile` is possible with a path.
So if I follow you, it may be an error in AWS API that has a wrong regexp to check instance profile has a valid name ?
